File: app/src/main/java/com/quickhandslogistics/utils/LanguageManager.java
package com.quickhandslogistics.utils;

import android.app.Activity;
import android.content.res.Configuration;
import android.os.Build;

import java.util.Locale;

public class LanguageManager {

    public enum Languages {
        ENG("en-us"),
        ES("es");
        private String language;

        Languages(String lang) {
            language = lang;
        }
        @Override
        public String toString() {
            return language;
        }
    }

    public static void setLanguage(Activity activity, String language) {
        Locale locale = new Locale(language);
        Locale.setDefault(locale);
        Configuration config = new Configuration();
        if (Build.VERSION.SDK_INT >= 17) {
            config.setLocale(locale);
        } else {
            config.locale = locale;
        }
        activity.getBaseContext().getResources().updateConfiguration(config, activity.getBaseContext().getResources().getDisplayMetrics());
    }
}

File: app/src/main/java/com/quickhandslogistics/utils/LanguageManager.kt
package com.quickhandslogistics.utils

import android.app.Activity
import android.content.res.Configuration
import com.franmontiel.localechanger.LocaleChanger
import java.util.*

object LanguageManager {
    fun setLanguage(activity: Activity, language: String?) {
        LocaleChanger.setLocale(Locale(language))
        val locale = Locale(language)
        Locale.setDefault(locale)
        val config = Configuration()
        config.setLocale(locale)
        activity.baseContext.resources.updateConfiguration(config, activity.baseContext.resources.displayMetrics)
    }

    enum class Languages(private val language: String) {
        ENG(AppConstant.LANGUAGE_ENGLISH_CODE), SPANISH(AppConstant.LANGUAGE_SPANISH_CODE);

        override fun toString(): String {
            return language
        }
    }
}